scripts/python/qa-scripts/createVCAAggregate.py

```diff
@@ -0,0 +1,523 @@
+from gspreadWrapper import GspreadWrapper
+from options import Options
+from utils import Utils
+
+from gspread.models import Cell
+from gspread_formatting import *
+
+from time import sleep
+import pandas as pd
+import json
+import os
+
+class createVCAAggregate():
+    def __init__(self):
+        self.opt = Options()
+        self.utils = Utils()
+        self.gspreadWrapper = GspreadWrapper()
+        self.proposals = json.load(open('proposals.json'))
+        self.vcas = json.load(open('vcas.json'))
+        self.vcasFiles = []
+
+        self.badColumns = [self.opt.notValidCol]
+        self.goodColumns = [self.opt.goodCol]
+        self.excellentColumns = [self.opt.excellentCol]
+        self.allColumns = self.badColumns + self.goodColumns + self.excellentColumns
+
+        self.dChallenges = self.opt.distinctChallenges
+
+
+    def prepareBaseData(self):
+        self.gspreadWrapper.getVCAMasterData()
+        self.vcaMerged = pd.DataFrame(columns=self.gspreadWrapper.dfVca.columns, data=None)
+        self.vcaMerged['name'] = ''
+        self.dfVca = self.gspreadWrapper.dfVca.set_index('id')
+        self.gspreadWrapper.getProposersMasterData()
+        self.dfMasterProposers = self.gspreadWrapper.dfMasterProposers.set_index('id')
+        # Set all counters to 0
+        self.dfVca[self.opt.noVCAReviewsCol] = 0
+        for col in self.allColumns:
+            self.dfVca[col] = 0
+            self.dfVca['Result ' + col] = 0
+
+    def prepareVCAsFileList(self):
+        for currentDirPath, currentSubdirs, currentFiles in os.walk('./vcas-files'):
+            for aFile in currentFiles:
+                if aFile.endswith(".csv") :
+                    fpath = str(os.path.join(currentDirPath, aFile))
+                    self.vcasFiles.append(fpath)
+
+    def loadVCAsFiles(self):
+        self.prepareBaseData()
+        self.prepareVCAsFileList()
+        self.vcasData = []
+        self.vcasFileList = []
```

Comment:
same as before, is `vcasFileList` redundant?

Reply:
In this case the `vcasFileList` is not redudant because it's possible that some files from `vcasFiles` are not used (loading errors, vcas not present in the proper file).